### PR TITLE
Feature/add migration

### PR DESCRIPTION
**Pull request by C0Rs1K**
Create DbContext and models, add migration

### CODE REVIEW

File: Curtains/Curtains.Infrastructure/Models/Bracing.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Curtains.Infrastructure.Models
```
Comment (zd-prog) [inline Curtains/Curtains.Infrastructure/Models/Bracing.cs]: Models folder should be in Domain project

File: Curtains/Curtains.Infrastructure/Models/CurtainsContext.cs [hunk 2]
```diff
@@ -0,0 +1,351 @@
+﻿using System;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata;
+
+namespace Curtains.Infrastructure.Models
```
Comment (zd-prog) [inline Curtains/Curtains.Infrastructure/Models/CurtainsContext.cs]: Context should be in Data folder in Domain project.

**Review by zd-prog — CHANGES_REQUESTED** [review]

**Review by zd-prog — APPROVED** [review]

File: Curtains/Curtains.Domain/Data/CurtainsContext.cs [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+        public virtual DbSet<Role> Roles { get; set; } = null!;
+        public virtual DbSet<User> Users { get; set; } = null!;
+        public virtual DbSet<UserOrder> UserOrders { get; set; } = null!;
+
+        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+        {
+            if (!optionsBuilder.IsConfigured)
+            {
+#warning To protect potentially sensitive information in your connection string, you should move it out of source code. You can avoid scaffolding the connection string by using the Name= syntax to read it from configuration - see https://go.microsoft.com/fwlink/?linkid=2131148. For more guidance on storing connection strings, see http://go.microsoft.com/fwlink/?LinkId=723263.
+                optionsBuilder.UseSqlServer("Server=(LocalDB)\\MSSQLLocalDB;Database=Curtains;Trusted_Connection=true;");
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Data/CurtainsContext.cs]: Please create config file for this needs like as appsettings.json

File: Curtains/Curtains.Domain/Data/CurtainsContext.cs [hunk 4]
```diff
@@ -0,0 +1,351 @@
...
+            }
+        }
+
+        protected override void OnModelCreating(ModelBuilder modelBuilder)
+        {
+            modelBuilder.UseCollation("Cyrillic_General_CI_AS");
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Data/CurtainsContext.cs]: looks like this strig we can add to `appsettings.json`

File: Curtains/Curtains.Domain/Data/CurtainsContext.cs [hunk 5]
```diff
@@ -0,0 +1,351 @@
...
+
+            modelBuilder.Entity<Bracing>(entity =>
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Data/CurtainsContext.cs]: Are you sure that we need to describe every entity in this way? If we will have 100 entities => we will have file with 2000 lines of code. Looks like it's worst way,isn't it?
Better use ` modelBuilder.ApplyConfiguration()`
This [article](https://digitaldrummerj.me/ef-core-split-model-config/) can help you with it.

File: Curtains/Curtains.Domain/Data/CurtainsContext.cs (hunk 3 above)
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Data/CurtainsContext.cs]: We need to create `appsettings.json` for it.

File: Curtains/Curtains.Domain/Models/Bracing.cs [hunk 6]
```diff
@@ -0,0 +1,20 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Curtains.Infrastructure.Models
+{
+    public partial class Bracing
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Models/Bracing.cs]: Please rename all models to template [name]Model

File: Curtains/Curtains.Infrastructure/Migrations/20220901113709_InitCreate.Designer.cs [hunk 7]
```diff
@@ -0,0 +1,702 @@
...
+        {
+#pragma warning disable 612, 618
+            modelBuilder
+                .UseCollation("Cyrillic_General_CI_AS")
+                .HasAnnotation("ProductVersion", "6.0.8")
+                .HasAnnotation("Relational:MaxIdentifierLength", 128);
+
+            SqlServerModelBuilderExtensions.UseIdentityColumns(modelBuilder, 1L, 1);
+
+            modelBuilder.Entity("Curtains.Infrastructure.Models.Bracing", b =>
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Infrastructure/Migrations/20220901113709_InitCreate.Designer.cs:28]: Is it possible to divide migrations by entities with logical flow?

**Review by EgorAntonovich — CHANGES_REQUESTED** [review]

**Comment by EgorAntonovich** [comment]
[[BE] DB creation](https://trello.com/c/ioMnVGc3/47-be-db-creation)

File: Curtains/Curtains.Domain/Data/Maps/BracingMap.cs [hunk 8]
```diff
@@ -0,0 +1,23 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Metadata.Builders;
+
+namespace Curtains.Infrastructure.Models
+{
+    internal class BracingMap : IEntityTypeConfiguration<BracingModel>
```
Comment (EgorAntonovich) [inline Curtains/Curtains.Domain/Data/Maps/BracingMap.cs]: Rename Maps to Mapping[s] in whole project

**Review by EgorAntonovich — CHANGES_REQUESTED** [review]

**Review by EgorAntonovich — APPROVED** [review]
Looks good

**Comment by EgorAntonovich** [comment]
Please resolve all conversation till merge

**Comment by C0Rs1K** [comment]
[[BE] E-1 S-1 T-6 Create EF layer for connection between API & DB](https://trello.com/c/F1wpmUz4/49-be-e-1-s-1-t-6-create-ef-layer-for-connection-between-api-db)

File: Curtains/Curtains.Infrastructure/Migrations/20220901113709_InitCreate.Designer.cs [hunk 9]
```diff
@@ -0,0 +1,702 @@
+﻿// <auto-generated />
+using System;
+using Curtains.Infrastructure.Models;
```
Comment (davidrabko) [inline Curtains/Curtains.Infrastructure/Migrations/20220901113709_InitCreate.Designer.cs:3]: There seems to be an issue, maybe we should change this dependency to <code>using Curtains.Domain.Models;</code>

File: Curtains/Curtains.Infrastructure/Migrations/CurtainsContextModelSnapshot.cs [hunk 10]
```diff
@@ -0,0 +1,700 @@
+﻿// <auto-generated />
+using System;
+using Curtains.Infrastructure.Models;
```
Comment (davidrabko) [inline Curtains/Curtains.Infrastructure/Migrations/CurtainsContextModelSnapshot.cs:3]: There seems to be an issue, maybe we should change this dependency to using Curtains.Domain.Models;

File: Curtains/Curtains.Domain/Data/CurtainsContext.cs [hunk 11]
```diff
@@ -0,0 +1,64 @@
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+
+namespace Curtains.Domain.Models
```
Comment (davidrabko) [inline Curtains/Curtains.Domain/Data/CurtainsContext.cs:4]: correct the namespace - <code>Curtains.Domain.Data</code>

Comment (EgorAntonovich) [reply Curtains/Curtains.Domain/Data/CurtainsContext.cs:4]: good catch

**Pull request merged**